### PR TITLE
HIP-260: Smart Contract Traceability

### PR DESCRIPTION
**Pull request by shemnon**
**Description**:

HIP-260 changes to track smart contract object state chages

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by shemnon** [comment]
We have decided to remove ContractActions.  We are investigating if `stateChanges` will remain or if the whole PR will be closed.

File: services/contract_call_local.proto [hunk 1]
```diff
@@ -150,3 +156,51 @@ message ContractCallLocalResponse {
...
+    oneof writeMode {
+        /**
+         * The new value written to the slot.  Up to 32 bytes, big-endian, zero bytes left trimmed.
+         */
+        bytes valueWritten = 3;
+
+        /**
+         * If the value was only read this will be true, in lieu of valueWritten.
+         */
+        bool readOnly = 4;
```
Comment (steven-sheehy) [inline services/contract_call_local.proto:204]: This doesn't match the format we agreed upon in the HIP

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: services/contract_call_local.proto (hunk 1 above)
Comment (shemnon) [reply services/contract_call_local.proto:204]: Yep, grabbed the wrong cached files.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Pull request merged**